src/HalResource.php

```diff
@@ -360,7 +360,7 @@ private function aggregateEmbeddedCollection(string $name, array $collection, st
      * Exists as array_shift is destructive, and we cannot necessarily know the
      * index of the first element.
      *
-     * @param null|HalResource[] $resources
+     * @param array $resources
```

Comment:
This is the same information like in the method signature therefore it can be omitted.

Reply:
I'm a bit surprised that Psalm did that...